skt/kernelbuilder.py

```diff
@@ -27,19 +27,53 @@
 from threading import Timer
 
 
+def verify_target(target):
+    """
+    Verify the given target is something we support.
+
+    Args:
+        target:    A build target archtecture
+
+    Returns:
+        The value of the ARCH env and the config file variant as a tuple
+    """
+    if not target:
+        target = platform.machine()
+
+    if re.match('x86_64', target):
+        return ('x86_64', target)
+    elif re.match('s390x', target):
+        return ('s390', target)
+    elif re.match('ppc64', target):
+        return ('powerpc', target)
+    elif re.match('ppc64le', target):
+        return ('powerpc', target)
+    elif re.match('aarch64', target):
+        return ('arm64', target)
+
+    return (None, None)
+
+
 class KernelBuilder(object):
     def __init__(self, source_dir, basecfg, cfgtype=None,
                  extra_make_args=None, enable_debuginfo=False,
-                 rh_configs_glob=None):
+                 target=None):
         self.source_dir = source_dir
         self.basecfg = basecfg
         self.cfgtype = cfgtype if cfgtype is not None else "olddefconfig"
         self._ready = 0
         self.buildlog = "%s/build.log" % self.source_dir
         self.make_argv_base = ["make", "-C", self.source_dir]
         self.enable_debuginfo = enable_debuginfo
-        self.build_arch = self.get_build_arch()
-        self.rh_configs_glob = rh_configs_glob
+        self.target = target
+
+        (build_arch, config_arch) = verify_target(target)
+        if build_arch is None:
+            raise Exception("Unsupported target: %s" % target)
```

Comment:
Raising a generic exception and not a custom one makes harder catch it later. It'd be more readable if you create a custom exception, e.g.
```python
class UnsupportedTarget(Exception):
   "Raised when target is not supported"
```

Reply:
Also, I think it'd be better if you raise the exception inside the function to avoid repeated code.